### PR TITLE
Handle dos 8.3 short path notation

### PR DESCRIPTION
**Pull request by sadaaithal**
Fixes dos short path issue from https://github.com/Eugene-Mark/bigdata-file-viewer/issues/43

### CODE REVIEW

**Review by Eugene-Mark — CHANGES_REQUESTED** [review]

File: src/main/java/org/eugene/controller/Renderer.java [hunk 1]
```diff
@@ -85,11 +91,43 @@ public boolean loadAndShow(){
...
         Path path = new Path(absolutePath);
         return load(path);
     }
+    
+    String resolveWindowsShortPath(String input) {
+		if (SystemUtils.IS_OS_WINDOWS) {
+			if (Paths.get(input).toAbsolutePath().toString().contains("~")) {
+				java.nio.file.Path inputPath = Paths.get(input);
+				String psDosToLongPathCmdFmt = "powershell \"(Get-Item -LiteralPath '" + inputPath.toAbsolutePath().toString() +"').FullName\"";
+				System.out.println("sada-fix: dos 8.3 input path: " + input);
```
[the remark this answers is not in the document]
Comment (sadaaithal) [reply src/main/java/org/eugene/controller/Renderer.java:105]: will do. im new to github commits so looks like I committed my workspace changes as-is without these SOPs removed.

File: src/main/java/org/eugene/controller/Renderer.java [hunk 2]
```diff
@@ -1,151 +1,186 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class Renderer {
+
+	private Stage stage;
```
Comment (Eugene-Mark) [inline src/main/java/org/eugene/controller/Renderer.java:31]: Please set the indent to 4 spaces, instead of 8 spaces.

Comment (sadaaithal) [reply src/main/java/org/eugene/controller/Renderer.java:31]: apologies - im still an eclipse person where the defaults are in tabs. 
il change this to 4 spaces and commit

Comment (Eugene-Mark) [reply src/main/java/org/eugene/controller/Renderer.java:31]: Thanks, but seems the tab is still 8 spaces. The correct indent should look like this: 
image

Comment (sadaaithal) [reply src/main/java/org/eugene/controller/Renderer.java:31]: Apologies - I have now formatted it using https://github.com/google/google-java-format
Here is how it looks (before vs after )

image

I will commit this now.
Thanks again!

Comment (Eugene-Mark) [reply src/main/java/org/eugene/controller/Renderer.java]: Hi @sadaaithal , thanks for your effort to introduce google java format. However, the project now is not following it, can you simply just set your editor to make tab equals 4 spaces. As for eclipse, you can refer to [this link](https://stackoverflow.com/questions/407929/how-do-i-change-eclipse-to-use-spaces-instead-of-tabs)  about how to do that.

Comment (sadaaithal) [reply src/main/java/org/eugene/controller/Renderer.java]: hi @Eugene-Mark  ive made the change and it finally reflects the 4 spaced tab. 
Please check and hopefully this works.
Apologies for the back and forth  as im new to github 

best
-s

**Review by Eugene-Mark — APPROVED** [review]
Thanks for your contribution to the project! LGTM!

**Pull request merged**